### PR TITLE
Add Tests and TravisCI integration

### PR DESCRIPTION
**Pull request by thraxil**
I have plans to make some changes here. So this adds a basic test suite that is run by travis and uses coveralls for coverage tracking.

I've got pretty good coverage of the important parts of the code. The `except CalledProcessError as e` block I have been unable to test though. AFAICT, it will only ever be raised when `subprocess.Popen()` is called with `check=True` and we are not passing in that argument. If there's some other way it can be raised, I haven't been able to duplicate it.

### CODE REVIEW

**Comment by bezidejni** [comment]
@thraxil I see that you're still adding commits, just let me know when it's ready to be reviewed and I'll hop right on it.

**Comment by thraxil** [comment]
@bezidejni review it any time you want.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+all:
+	virtualenv .test-ve
```
Comment (bezidejni) [inline Makefile]: It might be good to specify the exact major version of python so we don't end up using Python 3.

**Review by bezidejni — CHANGES_REQUESTED** [review]
Just a small change, otherwise looks good.

**Comment by thraxil** [comment]
Still waiting on this one.

**Comment by bezidejni** [comment]
@thraxil Sorry about it, I didn't see that you made a change and I forgot about this PR.

**Review by bezidejni — APPROVED** [review]

**Pull request merged**